src/main/kotlin/br/com/foodcommander/controller/ProductController.kt

```diff
@@ -0,0 +1,24 @@
+package br.com.foodcommander.controller
+
+import br.com.foodcommander.controller.dto.ProductRequest
+import br.com.foodcommander.models.Product
+import br.com.foodcommander.service.ProductService
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/v1/products")
+class ProductController(val service: ProductService) {
+
+    @PostMapping
+    fun postProduct(@RequestBody productRequest: ProductRequest): ResponseEntity<Product> {
+        val product = service.createProduct(
+                Product(null, productRequest.name, productRequest.description))
```

Comment:
ao invés de passar null na instanciação do objeto pode no model atribuir null ao valor, assim não precisando passar o null na criação do objeto

Reply:
utilizar Named Parameter

```suggestion
                Product(name = productRequest.name, description = productRequest.description))
```